newa/cli.py

```diff
@@ -206,75 +206,153 @@ def cmd_event(ctx: CLIContext, errata_ids: list[str]) -> None:
-                print(
-                    f'* Would kick automated job for issue {action.type.name}'
-                    f'based on recipe from {action.job_recipe}:')
-                print()
-
-                jira_job = JiraJob(event=erratum_job.event,
-                                   erratum=erratum_job.erratum,
-                                   jira=issue,
-                                   recipe=Recipe(url=action.job_recipe))
-                ctx.save_jira_job('jira-', jira_job)
+            # Find existing issues related to erratum_job and action
+            search_result = jira.get_open_issues(action, all_respins=True)
+
+            # Issues related to the curent respin and previous one(s).
+            new_issues: list[Issue] = []
+            old_issues: list[Issue] = []
+            for jira_issue_key, jira_issue in search_result.items():
+                ctx.logger.info(f"Checking {jira_issue_key}")
+
+                # In general, issue is new (relevant to the current respin) if it has newa_id
+                # of this action in the description. Otherwise, it is old (relevant to the
+                # previous respins).
+                #
+                # However, it might happen that we encounter subtask issue that is new but its
+                # original parent task got dropped (by human mistake, newa would never do that).
+                # By this time new parent task already exists. Unfortunately, Jira REST API does
+                # not allow updating 'parent' field [1] and hence we cannot re-use the issue with
+                # updated parent - we need to handle it as an old one (unless it has KEEP on_respin
+                # action it will get dropped and new one is created with the proper parent).
+                #
+                # [1] https://jira.atlassian.com/browse/JRASERVER-68763
+                decision_result = ""
+                desc = jira_issue["description"]
+                if jira.newa_id(action) in desc:
+                    decision_result = "new"
+                    if action.type == IssueType.SUBTASK \
+                            and action.parent_id \
+                            and processed_actions[action.parent_id].id != jira_issue["parent"]:
+                        decision_result = "old"
+                else:
+                    decision_result = "old"
+
+                if decision_result == "new":
+                    new_issues.append(Issue(jira_issue_key))
+                else:
+                    old_issues.append(Issue(jira_issue_key))
+
+            # Old issue(s) can be re-used for the current respin.
+            if old_issues and action.on_respin == OnRespinAction.KEEP:
+                new_issues.extend(old_issues)
+                old_issues = []
+
+            # Processing new issues.
+            #
+            # 1. Either there is no new issue (it does not exist yet - we need to create it).
+            if len(new_issues) == 0:
```

Comment:
Python idiom for testing emptiness of a list: `if not new_issues`

Reply:
Fixed.